src/Discord.Net.WebSocket/DiscordSocketClient.Events.cs

```diff
@@ -59,12 +59,12 @@ public partial class DiscordSocketClient
             remove { _messageReceivedEvent.Remove(value); }
         }
         private readonly AsyncEvent<Func<SocketMessage, Task>> _messageReceivedEvent = new AsyncEvent<Func<SocketMessage, Task>>();
-        public event Func<ulong, Optional<SocketMessage>, Task> MessageDeleted
+        public event Func<ulong, SocketChannel, Optional<SocketMessage>, Task> MessageDeleted
```

Comment:
MessageCreated and MessageUpdated receive `SocketMessage` objects which have a `ISocketMessageChannel Channel` property. I believe that is the reason why only UserIsTyping receives the `ISocketMessageChannel` object directly?

I mean it doesn't really matter because you can cast it into any of the required objects, or even into the interface directly. Just seemed more appropriate?